Src/Services/ApiService.lua

```diff
@@ -158,25 +158,31 @@ function ApiService.RequestAsync(self: ApiService, apiMethod: "GET" | "POST", ap
 				table.insert(errorObject.Errors, decodedJson.message)
 			end
 
-			self.Reporter:Warn(`'{apiMethod}' request failed for API '{api}': %s`, errorObject)
+			self.Reporter:Warn(`'{apiMethod}' request failed for endpoint '{apiEndpoint}': %s`, errorObject)
 
 			reject(errorObject)
 		end)
 	end)
 end
 
-function ApiService.GetAsync(self: ApiService, api: string, data: { [any]: any })
-	return self:RequestAsync("GET", api, data)
+function ApiService.GetAsync(self: ApiService, apiEndpoint: string, queries: { [string]: any })
+	local url = apiEndpoint
+
+	for queryName, queryValue in queries do
+		url ..= `?{HttpService:UrlEncode(queryName)}={HttpService:UrlEncode(queryValue)}`
```

Comment:
If there are multiple then the first uses ? yes but the following use &

Reply:
good catch, will update :+1: